src/stack/Stack.java

```diff
@@ -5,30 +5,29 @@
 
 
 
-public class Stack {
+public class Stack<T> {
 
     static final int DEFAULT_SIZE = 16;
-    private int[] data;
+    private T[] data;
     private int tos; //index which determines the top of stack
 
     public Stack() {
-
         this(DEFAULT_SIZE);
     }
 
     public Stack(int size) {
-        data = new int[size];
+        this.data = (T[])new Stack[size];
         this.tos = -1;
     }
 
     /**
      * Add element into stack
      */
-    public void push(int value) throws StackOverflowError {
+    public void push(T value) throws StackOverflowError {
         if (this.tos == data.length - 1) {
             throw new StackOverflowError();
```

Comment:
write message in exception
 throw new StackOverflowError("Stack is already full");